homeassistant/components/conversation/__init__.py

```diff
@@ -50,6 +49,14 @@ def async_set_agent(hass: core.HomeAssistant, agent: AbstractConversationAgent):
     hass.data[DATA_AGENT] = agent
 
 
+def get_agent(hass: core.HomeAssistant) -> AbstractConversationAgent:
+    """Get agent."""
+    agent = hass.data.get(DATA_AGENT)
+    if agent is None:
+        agent = hass.data[DATA_AGENT] = DefaultAgent(hass)
+    return agent
+
+
 async def async_setup(hass, config):
     """Register the process service."""
 
```

Comment:
If you are able to extract `get_agent` to be able to get the config passed to `async_setup`. Could you also move `async def process` out of `async_setup` and have it use the new get_agent helper ?